kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -316,20 +340,23 @@ public Map<TopicPartition, OffsetAndMetadata> preCommit(
 
   @Override
   public void close(Collection<TopicPartition> partitions) {
-    for (TopicPartition tp : topicPartitionWriters.keySet()) {
+    for (TopicPartition tp : partitions) {
       try {
         topicPartitionWriters.get(tp).close();
       } catch (ConnectException e) {
         log.error("Error closing writer for {}. Error: {}", tp, e.getMessage());
       }
+      topicPartitionWriters.remove(tp);
     }
-    topicPartitionWriters.clear();
```

Comment:
This change is needed for support using `CooperativeStickyAssigner` in order to reduce the number of revoked partitions.